lib/matplotlib/backend_bases.py

```diff
@@ -2298,32 +2297,32 @@ def get_window_title(self):
             return self.manager.get_window_title()
 
     def set_window_title(self, title):
-        """
-        Set the title text of the window containing the figure.  Note that
-        this has no effect if there is no window (e.g., a PS backend).
+        """Set the title text of the window containing the figure.
+
+        This has no effect if there is no window (e.g., a PS backend).
         """
         if hasattr(self, "manager"):
             self.manager.set_window_title(title)
 
     def get_default_filename(self):
-        """
-        Return a string, which includes extension, suitable for use as
-        a default filename.
-        """
-        default_basename = self.get_window_title() or 'image'
-        default_basename = default_basename.replace(' ', '_')
+        """Return a suitable default filename, including the extension.
+        """
+        default_basename = self.get_window_title() or "image"
+        # Characters to be avoided in a NT path:
+        # https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx#naming_conventions
+        removed_chars = \
+            {"posix": " /\0", "nt": r'<>:"/\|?*\0'}.get(os.name, "_")
```

Comment:
The argument from the slides is
> new to regular expressions and globbing? be kind to
yourself and avoid
> - spaces in file names
> - punctuation
> - accented characters
> - different files named “foo” and “Foo”

but I don't see how this is even relevant, because globbing occurs after word splitting (and, of course, languages other than shell are not affected at all).
```
sh-4.4$ mkdir test; echo 'foo' > 'test/with space and àccént & punc;tu:a>t<ion'; cat test/*
foo
sh-4.4$ python -c 'import glob; print(glob.glob("test/*"))'
['test/with space and àccént & punc;tu:a>t<ion']
```